pluggy.py

```diff
@@ -477,14 +489,30 @@ def _verify_hook(self, hook, hookimpl):
                 "Plugin %r\nhook %r\nhistoric incompatible to hookwrapper" %
                 (hookimpl.plugin_name, hook.name))
 
-        for arg in hookimpl.argnames:
-            if arg not in hook.argnames:
-                raise PluginValidationError(
-                    "Plugin %r\nhook %r\nargument %r not available\n"
-                    "plugin definition: %s\n"
-                    "available hookargs: %s" %
-                    (hookimpl.plugin_name, hook.name, arg,
-                    _formatdef(hookimpl.function), ", ".join(hook.argnames)))
+        # positional arg checking
+        hian, han = set(hookimpl.argnames), set(hook.argnames)
+        notinspec = hian - han
+        notinimpl = han - hian
+
+        if notinspec:
+            raise PluginValidationError(
+                "Plugin %r for hook %r\nhookimpl definition: %s\n%s"
+                "Positional args {0} are declared in the hookimpl but "
```

Comment:
Oh shoot I left in the format string syntax by accident...
The only thing I wanted to improve was reporting **all** the incompatible args in a single error instead of requiring that user runs multiple times to figure them all out.